python/examples/explore.py

```diff
@@ -0,0 +1,296 @@
+"""
+Chemiscope.explore example
+==========================
+This example illustrates usage of different methods of dimensionality reduction and its
+visualisation with `chemiscope.explore`. The primary goal of this function is to enable
+quick and effective exploration of a dataset.
+
+To use this function, some additional dependencies are required. You can install them
+with the following command:
+
+.. code:: bash
+
+    pip install chemiscope[explore]
+
+In this example, we will explore several use cases, starting from basic applications to
+more customized scenarios.
+
+First, let's import the necessary packages that will be used throughout the examples.
+"""
+
+# %%
+import os
+
+import ase.io
+import numpy as np
+
+import chemiscope
+
+# %%
+#
+# Basic usage
+# +++++++++++
+#
+# This example shows the basic usage of the `chemiscope.explore`. At first, read or
+# load the structures from the dataset. Here we use an `ASE package
+# <https://wiki.fysik.dtu.dk/ase>`_ to read the structures from the file and have the
+# frames as the `ase.Atoms <https://wiki.fysik.dtu.dk/ase/ase/atoms.html>`_ objects.
+
+frames = ase.io.read("data/explore_c-gap-20u.xyz", ":")
+
+
+# %%
+#
+# Provide the frames to the `chemiscope.explore`. It will generate a Chemiscope
+# interactive widget with the reduced dimentionality of data.
+
+chemiscope.explore(frames)
+
+# %%
+#
+# In this basic case, no featurizer function is provided, so `chemiscope.explore` uses a
+# default method that applies `SOAP (Smooth Overlap of Atomic Positions)
+# <https://singroup.github.io/dscribe/latest/tutorials/descriptors/soap.html>`_ to
+# compute atomic structure descriptors and then performs `PCA (Principal Component
+# Analysis)
+# <https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html>`_
+# for dimensionality reduction. The resulting components are then added to the
+# properties to be used in visualization.
+
+
+# %%
+#
+# Example with SOAP and KPCA
+# ++++++++++++++++++++++++++
+#
+# This part illustrates how to create a custom function for dimensionality reduction
+# as an argument (`featurize`) to `chemiscope.explore`. Inside this function, we perform
```

Comment:
```suggestion
# as an argument (``featurize``) to `chemiscope.explore`. Inside this function, we perform
```

RestructuredText uses double backtick for inline code (while markdown uses a single one). Could you update everything to with the right syntax?

Reply:
Done